### PR TITLE
Slack integration

### LINKED ISSUE #211: Slack Integration
**Endpoints to be supported:** 

`GET https://api.revert.dev/chat/channels`: List all channels of the Slack workspace connected

`GET https://api.revert.dev/chat/users`: List all users of the Slack workspace connected 

`POST https://api.revert.dev/chat/message`: Create a message in a specific channel

### PR DESCRIPTION
**Pull request by ashutoshdhande**
### Description

Adds Slack integration with apis to retrieve and send messages on Slack workspaces 

Closes #211 

### Type of change

-   [X] New feature (non-breaking change which adds functionality)
-   [X] This change requires a documentation update

### Video Demo
- [Oauth using react client](https://www.loom.com/share/1914ff8264094d86898d0ebdab184d64?sid=d53c5ab3-c646-4f7c-acf2-12ba247d4091)
- [Endpoints](https://www.loom.com/share/57354696a2044916918bb9b884546e88?sid=2ca374cd-7a38-4a3e-83cb-4de1c7d2afc8)

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test setup

-   By using Postman
-   By using react-app from port 3001

### Checklist:

-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my code
-   [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by jatinsandilya** [comment]
@ashutoshdhande Could you add the loom shared on discord here in the PR description as well? Thanks!

File: packages/backend/services/metadata.ts [hunk 1]
```diff
@@ -91,6 +91,15 @@ const metadataService = new MetadataService({
                     scopes: [],
                     clientId: '',
                 },
+                {
+                    integrationId: 'slack',
+                    name: 'slack',
+                    imageSrc:
+                        'https://w7.pngwing.com/pngs/345/302/png-transparent-chat-slack-slack-logo-social-media-icon-thumbnail.png',
```
Comment (jatinsandilya) [inline packages/backend/services/metadata.ts:98]: TODO: update this to a cloudinary link

File: packages/react/vite.config.ts [hunk 2]
```diff
@@ -2,10 +2,15 @@ import react from '@vitejs/plugin-react';
 import path from 'node:path';
 import { defineConfig } from 'vite';
 import dts from 'vite-plugin-dts';
+// import mkcert from 'vite-plugin-mkcert';
 
 export default defineConfig({
     server: {
         port: 3001,
+        https: {
```
Comment (jatinsandilya) [inline packages/react/vite.config.ts:10]: @ashutoshdhande Please remove the usage keys & certs in this PR including this file.

File: packages/react/vite.config.ts [hunk 3]
```diff
@@ -2,10 +2,15 @@ import react from '@vitejs/plugin-react';
 import path from 'node:path';
 import { defineConfig } from 'vite';
 import dts from 'vite-plugin-dts';
+// import mkcert from 'vite-plugin-mkcert';
```
Comment (jatinsandilya) [inline packages/react/vite.config.ts:5]: Please remove any commented code in this PR if it's not required

File: packages/backend/constants/common.ts [hunk 4]
```diff
@@ -34,6 +34,7 @@ export const DEFAULT_SCOPE = {
     [TP_ID.zohocrm]: ['ZohoCRM.modules.ALL', 'ZohoCRM.settings.ALL', 'ZohoCRM.users.ALL', 'AaaServer.profile.READ'],
     [TP_ID.sfdc]: [], // https://help.salesforce.com/s/articleView?id=sf.remoteaccess_oauth_tokens_scopes.htm&type=5
     [TP_ID.pipedrive]: [],
+    [TP_ID.slack]: ['users:read', 'users.profile:read'],
```
Comment (jatinsandilya) [inline packages/backend/constants/common.ts:37]: Do we not need the `channels:read` or `channels:write` permission here?

Comment (ashutoshdhande) [reply packages/backend/constants/common.ts:37]: At least for retrieving channels it worked without those permissions.

**Pull request closed without merging**